drivers/clock_control/clock_control_nrf2_fll16m.c

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_nrf_fll16m
+
+#include "clock_control_nrf2_common.h"
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/clock_control/nrf_clock_control.h>
+#include <hal/nrf_lrcconf.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(clock_control_nrf2, CONFIG_CLOCK_CONTROL_LOG_LEVEL);
+
+BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 1,
+	     "multiple instances not supported");
+
+#define FLAG_HFXO_STARTED BIT(FLAGS_COMMON_BITS)
+
+#define FLL16M_MODE_OPEN_LOOP   0
+#define FLL16M_MODE_CLOSED_LOOP 1
+#define FLL16M_MODE_BYPASS      2
+#define FLL16M_MODE_DEFAULT     FLL16M_MODE_OPEN_LOOP
+
+#define FLL16M_LFXO_NODE DT_INST_PHANDLE_BY_NAME(0, clocks, lfxo)
+#define FLL16M_HFXO_NODE DT_INST_PHANDLE_BY_NAME(0, clocks, hfxo)
+
+#define FLL16M_HAS_LFXO \
+	(IS_ENABLED(FLL16M_LFXO_NODE) && DT_NODE_HAS_STATUS(FLL16M_LFXO_NODE, okay))
+
+#define FLL16M_LFXO_ACCURACY DT_PROP(FLL16M_LFXO_NODE, accuracy_ppm)
+#define FLL16M_HFXO_ACCURACY DT_PROP(FLL16M_HFXO_NODE, accuracy_ppm)
+#define FLL16M_OPEN_LOOP_ACCURACY DT_INST_PROP(0, open_loop_accuracy_ppm)
+#define FLL16M_CLOSED_LOOP_BASE_ACCURACY DT_INST_PROP(0, closed_loop_base_accuracy_ppm)
+#define FLL16M_MAX_ACCURACY FLL16M_HFXO_ACCURACY
+
+/* Closed-loop mode uses LFXO as source if present, HFXO otherwise */
+#if FLL16M_HAS_LFXO
+#define FLL16M_CLOSED_LOOP_ACCURACY (FLL16M_CLOSED_LOOP_BASE_ACCURACY + FLL16M_LFXO_ACCURACY)
+#else
+#define FLL16M_CLOSED_LOOP_ACCURACY (FLL16M_CLOSED_LOOP_BASE_ACCURACY + FLL16M_HFXO_ACCURACY)
+#endif
+
+/* Clock options sorted from lowest to highest accuracy */
```

Comment:
Now when I'm thinking about this array, it actually is sorted from the lowest to the highest power consumption.
In the case of this specific array, accuracy is also improving. But it's not necessarily the rule for all similar arrays.

We aim to sort these arrays by power consumption to make the algorithm select the mode with _the lowest power consumption_, which meets the requested accuracy and precision numbers.

Reply:
I will leave this for now, since the API clock spec requests set frequency, accuracy and precision. That's why these arrays focus on those properties for ranking